apps/ssr-tests-v9/src/utils/getImportsFromIndexFile.ts

```diff
@@ -8,15 +8,19 @@ type StoryImport = {
   path: string;
 };
 
-// TODO: Once will be solved, can be simplified https://github.com/microsoft/fluentui/issues/23393
 function getImportPrefix(filename: string): string {
-  const importPrefixByFile = path.basename(filename).split('.')[0];
-
-  if (importPrefixByFile === 'index') {
-    return path.basename(path.dirname(filename));
+  const importPrefix = path.basename(path.dirname(filename));
+
+  if (importPrefix === 'stories' || importPrefix[0].toLowerCase() === importPrefix[0]) {
+    throw new Error(
+      [
+        `The file "${filename}" has an incorrect location a directory name.`,
+        `All stories should be placed in "/src/stories/ComponentName" directory.`,
+      ].join(' '),
+    );
```

Comment:
A new check to assert that stories are not inside `stories` directory & directory name starts with a capital letter.